ReSharper.FSharp/src/FSharp.Psi/src/Resolve/FSharpParameterUtil.cs

```diff
@@ -63,5 +68,71 @@ public static class FSharpParameterUtil
       var field = unionCase.Fields.FirstOrDefault(f => f.Name == reference.GetName());
       return field?.GetDeclaredElement(referenceOwner.GetPsiModule(), referenceOwner);
     }
+
+    public static IEnumerable<IEnumerable<string>> GetParametersGroupNames(ITreeNode node) =>
```

Comment:
Please allow checking length and getting an item by index without enumerating the sequence multiple times.